### ISSUE TITLE
Explanation about Sanskrit kosha structures

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/COLOGNE#339]
Ref - https://github.com/sanskrit-lexicon/COLOGNE/issues/337#issuecomment-764195378 

> Incidentally, what are the TWO 'headwords?' आदित्य and रोचिस् under अंशु: . Are they the names of two
wordlists in which अंशु: occurs? Please provide a minor introduction regarding this kind of 'thesauras'.

This issue is devoted to explaining the structure of Sanskrit koshas to @funderburkjim 

There are practically two types of Sanskrit-Sanskrit koshas which have come down to us.

# 1. Synonymic dictionaries.

These are dictionaries which enlist all the synonyms for a given word at one go.
e.g. Abhidhānaratnamālā of Halāyudha

ब्रह्मा स्रष्टा परमेष्ठी धाता पद्मभूः सुरज्येष्ठः ।
वेधा विधिर्विरिञ्चो हिरण्यगर्भः शतानन्दः ॥ ६ ॥
शम्भुः स्वयम्भूर्द्रुहिणश्चतुर्वक्त्रः प्रजापतिः ।
पितामहो जगत्कर्ता विरञ्चिः कमलासनः ॥ ७ ॥

The following are the headwords contained in these two verses. 
They are all synonyms of each other.
ब्रह्मन् = स्रष्टृ = परमेष्ठिन् = धातृ = पद्मभू = सुरज्येष्ठ = वेधस् = विधि = विरिञ्च = हिरण्यगर्भ = शतानन्द = शम्भु = स्वयम्भू = द्रुहिण = चतुर्वक्त्र = प्रजापति = पितामह = जगत्कर्तृ = विरञ्चि = कमलासन

As far as I know there are no such dictionaries in the Western world. The closest to them are something known as thesaurus eġ. https://www.merriam-webster.com/thesaurus

# 2. Homonymic dictionaries

These dictionaries are like the regular dictionaries in the western world.
There is a headword, followed by its meaning.
e.g. Viśvalocanakośa of Śrīdharasena.

दीपको वागलङ्कारे प्रदीपे दीप्तिकारके ।

Here दीपक is the headword. वागलङ्कार, प्रदीप, दीप्तिकारक are the meaning.
In other words दीपक = वागलङ्कार, दीपक = प्रदीप, दीपक = दीप्तिकारक
But वागलङ्कार != प्रदीप, प्रदीप != दीप्तिकारक, वागलङ्कार != दीप्तिकारक

# 3. Mixed dictionaries

There  are a few dictionaries which have some chapters written in the synonymic style and some in homonymic style. 

Hope this clarifies your question.
If there are any further clarifications needed, I would be glad to help.

**Comment by gasyoun** [comment]
@drdhaval2785 what are the stats Synonymic vs. Homonymic in your list?

**Comment by drdhaval2785** [comment]
Difficult question. 
Synonymic are fewer in number, but quoted more.
homonymics are greater in number, but quoted less. Exact numbers - I am not sure.
If anyone is interested, he can check up ;nymic{} tag in metadata.

**Comment by funderburkjim** [comment]
Thanks for explanation.  

In the synonymic sanskrit-sanskrit dictionaries,  how are they used in pre-computer times?

For example, is there some ordering in such dictonaries that lets a reader 'look up' a given word in a synonymic dictionary?

**Comment by drdhaval2785** [comment]
There is no such ordering.
When a commentator on Sanskrit literature wants to specify the meaning, he quotes the whole verse from the thesaurus where this word occurs.

If I am to explain a line in literature, where the word 'परशुधर' occurs,
I would as a commentator quote 'हेरम्बो लम्बोदर आखुरथो गणपतिश्च गजवदनः । परशुधर एकदन्तो विनायको विघ्नराजश्च ॥ इति हलायुधः'
Or still shorter 'परशुधर एकदन्तो विनायको विघ्नराजश्च इति हलायुधः'

Because the reader knows at least one word गणपति / विनायक, he extrapolates that the word परशुधर also refers to the deity गणपति.

This is how the thesaura were use in pre-computer times.

It was mandatory for any Sanskrit scholars to have 'नामलिङ्गानुशासन' of अमरसिंह by heart.

**Comment by gasyoun** [comment]
> 'नामलिङ्गानुशासन' of अमरसिंह by heart.

Only it? 1 was enough by heart?

**Comment by drdhaval2785** [comment]
1 was and still is enough by heart.

**Comment by funderburkjim** [comment]
All of the words in the example refer to गणेश  acc. to MW.  Interesting that
the word गणेश is actually not one of the words in the verse.

Noticed one variance with MW, गणेश  is mentioned under विघ्नराज्  but
under  विघ्नराज in MW we see only id.   This illustrates that it would be useful
to 'resolve' the 'idems' in MW.   It also illustrates where the 'keydoc' notion
might be useful:  If someone looks up either विघ्नराज्   or विघ्नराज , he ideally
would get both entries; the document has two entries.

**Comment by funderburkjim** [comment]
According to [this reference](https://github.com/sanskrit-lexicon/COLOGNE/issues/336#issuecomment-773637958):
```
is divided into five kāṇḍas or sections as follows: 1. svarkāṇḍa, 2. bhūmikāṇḍa, 3. pātālakāṇḍa, 4. sāmānyakāṇḍa, and 5. anekārthāṇḍa. The first four of these deal with synonyms while the last is devoted to homonyms and the indeclinables. 
```
Does our digitization of armh have the fifth section -- If so, where does it start? (L=?) and did you have to handle it differently?

**Comment by gasyoun** [comment]
> This illustrates that it would be useful to 'resolve' the 'idems' in MW. 

You knew it ages ago )) Any clue how to?

**Comment by funderburkjim** [comment]
## question re  `<vn>` in armh metalines.

Example: `<L>6977<pc>0091<k1>KaRqaparaSu<k2>KaRqaparaSu<vn>5.1.1.788`

(Conjecture:  This is the first section 5 entry.  If so that answers my question
above).

This information does not carry into armh.xml,  and does not appear in the
display.  Perhaps it could carry into the xml as <info vn="5.1.1.788"/> and
then basicdisplay.php could be adjusted to display this.


There are several (8) records of armh.txt with this same 'vn':
```
<L>6977<pc>0091<k1>KaRqaparaSu<k2>KaRqaparaSu<vn>5.1.1.788
rudre'pi KaRqaparaSurvESravaRe'pyekakuRqalaH proktaH .
<LEND>
<L>6978<pc>0091<k1>rudra<k2>rudra<vn>5.1.1.788
rudre'pi KaRqaparaSurvESravaRe'pyekakuRqalaH proktaH .
<LEND>
<L>6979<pc>0091<k1>ekakuRqala<k2>ekakuRqala<vn>5.1.1.788
rudre'pi KaRqaparaSurvESravaRe'pyekakuRqalaH proktaH .
<LEND>
<L>6980<pc>0091<k1>vESravaRa<k2>vESravaRa<vn>5.1.1.788
rudre'pi KaRqaparaSurvESravaRe'pyekakuRqalaH proktaH .
<LEND>
<L>6981<pc>0091<k1>pratihAra<k2>pratihAra<vn>5.1.1.788
dvAre'pi pratihAraH prAkArAgre'pi kapiSIrzam .. 788 ..
<LEND>
<L>6982<pc>0091<k1>dvAra<k2>dvAra<vn>5.1.1.788
dvAre'pi pratihAraH prAkArAgre'pi kapiSIrzam .. 788 ..
<LEND>
<L>6983<pc>0091<k1>kapiSIrza<k2>kapiSIrza<vn>5.1.1.788
dvAre'pi pratihAraH prAkArAgre'pi kapiSIrzam .. 788 ..
<LEND>
<L>6984<pc>0091<k1>prAkArAgra<k2>prAkArAgra<vn>5.1.1.788
dvAre'pi pratihAraH prAkArAgre'pi kapiSIrzam .. 788 ..
<LEND>
```

**Comment by drdhaval2785** [comment]
> Conjecture: This is the first section 5 entry. If so that answers my question above

Yes. This is the first entry of the 5th kAnda.

> This information does not carry into armh.xml, and does not appear in the display. Perhaps it could carry into the xml

It should be carried into armh.xml . I am ready to do the same, if you wish. I did not want to break anything which was working on Cologne. Therefore, pruned the armh.xml to be as compliant to Cologne dicts as possible. Let me know where you want this information to be carried forward? which tag?
